src/neo/Consensus/ConsensusContext.cs

```diff
@@ -405,6 +423,23 @@ public void Reset(byte viewNumber)
             TransactionHashes = null;
             PreparationPayloads = new ConsensusPayload[Validators.Length];
             if (MyIndex >= 0) LastSeenMessage[MyIndex] = (int)Block.Index;
+
+            if (resetFutures)
+                ResetFuturePayloads();
+        }
+
+        public void ResetFuturePayloads()
```

Comment:
When will we need to reset the future payloads?

Reply:
Perhaps we do not need this, @erikzhang, you are correct.

Reply:
We need to reset in the beginning and when validators length is modified.

I tried to improve but did not find an easy way for now.